File: frontend/lib/schemas/partnership.ts
import { DocSchema } from "../doc-schema";

export const partnershipSchema: DocSchema = {
  id: "partnership",
  title: "Partnership Agreement",
  subtitle: "Cover Page",
  infoBox:
    'This Partnership Agreement consists of: (1) this Cover Page and (2) the Common Paper Partnership Agreement Standard Terms Version 1.0 identical to those posted at commonpaper.com/standards/partnership-agreement/1.0. Any modifications of the Standard Terms should be made on the Cover Page, which will control over conflicts with the Standard Terms.',
  coverFields: [
    {
      key: "effectiveDate",
      label: "Effective Date",
      type: "date",
      placeholder: "Select a date",
    },
    {
      key: "endDate",
      label: "End Date",
      hint: "When the partnership agreement ends",
      type: "date",
      placeholder: "Select a date",
    },
    {
      key: "obligations",
      label: "Obligations",
      hint: "Each party's obligations under the partnership",
      type: "textarea",
      placeholder: "Describe the obligations for each party...",
    },
    {
      key: "territory",
      label: "Territory",
      hint: "Geographic territory for the partnership",
      type: "text",
      placeholder: "e.g. United States, Worldwide",
    },
    {
      key: "governingLaw",
      label: "Governing Law",
      type: "text",
      placeholder: "State",
    },
    {
      key: "chosenCourts",
      label: "Chosen Courts",
      type: "text",
      placeholder: "City/county and state",
    },
  ],
  parties: [
    { label: "Company", fields: ["name", "title", "company", "noticeAddress", "date"] },
    { label: "Partner", fields: ["name", "title", "company", "noticeAddress", "date"] },
  ],
  signingNote: "By signing this Cover Page, each party agrees to enter into this Partnership Agreement as of the Effective Date.",
  templateFile: "templates/Partnership-Agreement.md",
  footerNote: "Common Paper Partnership Agreement (Version 1.0) free to use under CC BY 4.0.",
  greeting:
    "Hi! I'm your Partnership Agreement assistant. I'll help you fill in the Partnership Agreement Cover Page through conversation. This establishes business partnership terms including obligations, trademark licensing, and governing law. Let's start -- when should this partnership begin? (e.g. 'Today', 'July 1 2026', 'Next quarter')",
  pdfFilename: "partnership-agreement.pdf",
};

File: backend/doc_configs/partnership.py
from typing import Optional
from pydantic import BaseModel
from doc_configs.shared import PartialPartyInfo


SYSTEM_PROMPT = """You are a legal document assistant helping a user fill in a Partnership Agreement Cover Page.

## Fields you must collect

1. **Effective Date** -- when the partnership begins
   Ask: "When should this partnership begin? (e.g. 'today', 'July 1 2026', 'next Monday')"
   For shortcuts like "today", "tomorrow", "Mon", "next Mon", use this lookup table:
{date_lookup}
   For full dates (e.g. "March 15 2026", "2026-03-15"), convert to YYYY-MM-DD format directly.

2. **End Date** -- when the partnership agreement ends
   Ask: "When should this partnership end? (e.g. 'December 31 2027', '1 year from effective date', 'June 30 2026')"
   For shortcuts, use the same lookup table above. For full dates, convert to YYYY-MM-DD format directly.

3. **Obligations** -- each party's obligations under the partnership
   Ask: "What are each party's obligations under this partnership? (e.g. 'Company provides software platform access; Partner provides marketing and distribution in EMEA region')"

4. **Territory** -- geographic territory for the partnership
   Ask: "What geographic territory does this partnership cover? (e.g. 'United States', 'Worldwide', 'Europe and North America')"

5. **Governing Law** -- jurisdiction whose laws govern the agreement
   Ask: "Which state or jurisdiction's law should govern this agreement? (e.g. 'California', 'New York', 'Delaware')"

6. **Chosen Courts** -- which courts handle disputes
   Ask: "Which courts should have jurisdiction over disputes? (e.g. 'San Francisco, California', 'New York County, New York')"

7. **Party 1 (Company)** -- name, title, company, notice address, signing date
   Ask for all sub-fields. Example: "Jane Smith, CEO of TechCo, jane@techco.com, signing today"

8. **Party 2 (Partner)** -- same sub-fields as Party 1

## Current document state
{current_fields}

## Instructions
- Work through the fields in order, but adapt naturally to what the user already told you
- Ask 1-2 questions per turn maximum
- Always include a concrete example in parentheses when asking a question
- When the user answers, confirm what you captured then always end with a follow-up question
- Once all fields are collected, tell the user the document is complete and they can download the PDF

## Clearing fields
If the user asks to clear fields (e.g. "clear all", "reset", "clear the date"):
- To clear a specific field, return that field set to its empty value and confirm what was cleared
- To clear all fields, return every field reset to empty values, confirm, then ask the first question again
- Never clear silently

Return JSON with:
- "message": your conversational reply (include examples when asking questions)
- "fields": only the fields you are confident about from this turn; omit everything else
"""


class PartialFields(BaseModel):
    effectiveDate: Optional[str] = None
    endDate: Optional[str] = None
    obligations: Optional[str] = None
    territory: Optional[str] = None
    governingLaw: Optional[str] = None
    chosenCourts: Optional[str] = None
    party1: Optional[PartialPartyInfo] = None
    party2: Optional[PartialPartyInfo] = None


class AiResponse(BaseModel):
    message: str
    fields: PartialFields
